File: hw2/RaceMonitor_Java/src/Messages/StartedUpdate.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Messages;

import Exceptions.ApplicationException;
import Racedata.AthleteRaceStatus;
import java.time.LocalDateTime;
/**
 *
 * @author Shimin
 * @author Stephen Clyde
 */
public class StartedUpdate extends AthleteUpdate {
    private LocalDateTime officialStartTime;
    
    public StartedUpdate() {
        super(AthleteRaceStatus.Started);
    }
    
    public StartedUpdate(String[] properties) throws ApplicationException {
        super (AthleteRaceStatus.Started, properties);
        
        if (properties.length != 4 || !properties[0].equals(AthleteRaceStatus.Started.toString()))
            throw new ApplicationException("Invalid properties");

        officialStartTime = LocalDateTime.parse(properties[3], dateTimeFormatter);
    }
    
    @Override
    public String toString() { return super.toString() + "," + officialStartTime.format(dateTimeFormatter); }
    
    public void setOfficialStartTime (LocalDateTime value) {
        officialStartTime = value;
    }

    public LocalDateTime getOfficialStartTime () {
        return officialStartTime;
    }

}

File: hw2/RaceMonitor_CSharp/RaceData/Messages/StartedUpdate.cs
using System;

namespace RaceData.Messages
{
    public class StartedUpdate : AthleteUpdate
    {
        public DateTime OfficialStartTime { get; set; }

        public StartedUpdate() : base(AthleteRaceStatus.Started) { }

        public StartedUpdate(string[] properties) : base(AthleteRaceStatus.Started, properties)
        {
            if (properties.Length != 4 || properties[0] != AthleteRaceStatus.Started.ToString())
                throw new ApplicationException("Invalid properties");

            OfficialStartTime = Convert.ToDateTime(properties[3]);
        }

        public override string ToString()
        {
            return $"{base.ToString()},{OfficialStartTime}";
        }
    }
}
